### PR TITLE
Add missing 'skipped' option to `needs`

### PR DESCRIPTION
**Pull request by Aebrathia**
### Why:

I found out that the `needs.<job id>.result` context in Github Actions has an undocumented option `skipped` when the job dependency is skipped.

### What's being changed:

I added 'skipped' to the list of possible values.

### Check off the following:
- [x] All of the tests are passing.
- [x] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks for opening a PR @Aebrathia! I'll get this triaged for review ✨

File: content/actions/reference/context-and-expression-syntax-for-github-actions.md [hunk 1]
```diff
@@ -164,7 +164,7 @@ The `needs` context contains outputs from all jobs that are defined as a depende
 | `needs.<job id>` | `object` | A single job that the current job depends on. |
 | `needs.<job id>.outputs` | `object` | The set of outputs of a job that the current job depends on. |
 | `needs.<job id>.outputs.<output name>` | `string` | The value of a specific output for a job that the current job depends on. |
-| `needs.<job id>.result` | `string` | The result of a job that the current job depends on. Possible values are `success`, `failure`, or `cancelled`. |
+| `needs.<job id>.result` | `string` | The result of a job that the current job depends on. Possible values are `success`, `failure`, `cancelled` or `skipped`. |
```
Comment (lucascosti) [inline content/actions/reference/context-and-expression-syntax-for-github-actions.md:167]: Just needs a little oxford comma: 🙂 

```suggestion
| `needs.<job id>.result` | `string` | The result of a job that the current job depends on. Possible values are `success`, `failure`, `cancelled`, or `skipped`. |
```

**Review by lucascosti — APPROVED** [review]
@Aebrathia Thanks a lot for this PR!

It took me a little while to figure out if it was even possible for a job that `needs` a skipped job to actually run in order to see the context. but I did eventually get a workflow run that resulted in the output (using `if: ${{ always() }}` 😁) :

https://github.com/lucascosti/actions-public-playground/runs/1246235670?check_suite_focus=true#step:3:13 

I'll get this PR merged in 🚀

**Comment by Aebrathia** [comment]
@lucascosti thanks for your review, glad to know about Oxford commas.

The issue you faced is a bit tricky and needs a bit doc digging. Default `if:` value is `${{ success() && your_custom_if_condition }}` unless you explicitely use a [job status check function](https://docs.github.com/en/free-pro-team@latest/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions) (like I did below).

```yaml
name: Deploy
jobs:
  update:
    if: # should update deployment configuration
    steps:
      - run: # update project
  deploy:
    needs: update
    if: ${{ success() || needs.update.result == 'skipped' }} # Deploy even if no update (manual redeploy…)
    steps:
      - run: # deploy
```

**Comment by janiceilene** [comment]
@Aebrathia If you haven't already, you can add yourself to [the list of contributors](https://github.com/github/docs#contributors-) by creating a new comment in this PR using [these instructions](https://allcontributors.org/docs/en/bot/usage#commands). Thanks again! :sparkles:

**Comment by lucascosti** [comment]
>The issue you faced is a bit tricky and needs a bit doc digging. Default if: value is ${{ success() && your_custom_if_condition }} unless you explicitely use a job status check function (like I did below).

Ah, of course! Yes, I forgot about situation where `needs` can just serialize jobs, rather than only for strict dependencies. 😁

**Pull request merged**